microsoft-edge/extensions-chromium/developer-guide/minimize-page-load-time-impact.md

```diff
@@ -0,0 +1,251 @@
+
+
+<!-- ------------------------------ -->
+#### Programmatically load content scripts at runtime
+
+To programmatically load your content scripts at runtime, only when needed, use the `chrome.scripting` API . The `chrome.scripting` API provides more control over when and where your content script is loaded.
+
+For example, you can use the `chrome.scripting` API to load your content script only after the user has interacted with the webpage or extension UI, such as when clicking on an extension button, or a part of the webpage.
+
+If you use the `chrome.scripting` API when the user interacts with the webpage, make sure to carefully consider if you need to repeatedly load your content script every time the interaction happens. Loading content scripts too frequently can cause user experience issues or errors.
+
+
+<!-- ====================================================================== -->
+## Avoid blocking calls or long running synchronous tasks
+
+Blocking calls and long-running synchronous tasks can delay the webpage load or make other aspects of a webpage slower, as well as negatively impact the UI responsiveness.
+
+Blocking calls are JavaScript operations that prevent running other code until they complete. For example, using the `XMLHttpRequest`, `localStorage`, or `chrome.storage.sync` APIs, which are synchronous, prevents the webpage from running other code.
+
+Long running synchronous tasks are those that take a long time to complete, preventing the browser from running other webpage code while they are running. This can include complex calculations, loops, or string manipulations.
+
+Use asynchronous or non-blocking code, such as the Fetch API, JavaScript Promises, or Web Workers where possible. Asynchronous or non-blocking code allow the execution of other code while waiting for the completion of a task, without blocking browser process that runs the webpage.
+
+Note that, although using Web Workers to move your complex code logic to another thread is a good practice, it might still slow down devices that have a low CPU core count or that are already busy.
+
+Below is an example using the Fetch API. While the data is being fetched, the browser is not blocked and can run other code:
+
+```javascript
+// Asynchronously load data from a JSON file.
+fetch("data.json")
+  .then(response => response.json())
+  .then(data => {
+    // Do something with the data.
+  });
+```
+
+
+<!-- ------------------------------ -->
+#### Asynchronously store data
+
+To store data in your extension, use the `chrome.storage.local` API instead of the `localStorage` API, which is a synchronous API. The `chrome.storage.local` API is asynchronous  and can store and retrieve data more efficiently without affecting the performance of the webpage where your extension is running. For example, you can use the `chrome.storage.local.get` method to retrieve a previously stored value, and then use the result in a callback function:
```

Comment:
```suggestion
To store data in your extension, use the `chrome.storage.local` API instead of the `localStorage` API, which is a synchronous API. The `chrome.storage.local` API is asynchronous and can store and retrieve data more efficiently, without affecting the performance of the webpage where your extension is running. For example, you can use the `chrome.storage.local.get` method to retrieve a previously stored value, and then use the result in a callback function:
```